README.md

```diff
@@ -271,9 +271,23 @@ Note the cost and time depends on your job settings and the regions.
 ### Run the Beam pipeline with the Pub/Sub source
 When `INPUT_DATA` from the `.env` file defines a valid Pub/Sub topic (e.g., `projects/apache-beam-testing/topics/Imagenet_openimage_50k_benchmark`),
 the Beam pipeline is created using the Pub/Sub source with `FixedWindows` and switches to `beam.io.fileio.WriteToFiles` that supports the streaming pipeline.
-We use `shards=0` here since 0 shards is the recommended approach and Dataflow would decide how many files it should write.
+Note for this toy example, writing the predictions to a GCS bucket is not efficient since the file size is quite small with few bytes.
+In practice, you might tune up [the autoscaling options](https://cloud.google.com/dataflow/docs/guides/troubleshoot-autoscaling) to optimize the streaming pipeline performance.
 Note that the streaming job will run forever until it is canceled or drained.
 
+### Run the Beam pipeline with Dataflow Flex Templates
+If you prefer to package all your code into a custom container and allow users to easily access your Beam pipeline,
+Dataflow Flex Template could be handy to create and run a Flex Template job using Google Cloud CLI or Google Cloud console. (More benefits about templates are [here](https://cloud.google.com/dataflow/docs/concepts/dataflow-templates#benefits).)
```

Comment:
a big benefit is that flex templates produce a reproducible launch environment. If we build the flex template from the custom sdk container image, then the launch and runtime environment will be compatible, more details at:  https://beam.apache.org/documentation/sdks/python-pipeline-dependencies/#make-the-launch-environment-compatible-with-the-runtime-environment

Reply:
please check https://github.com/google/dataflow-ml-starter/pull/19